manifests/cli.pp

```diff
@@ -0,0 +1,92 @@
+# = Pulpcore command line interface
+#
+# This class installs the Pulpcore command line interface (CLI).
+#
+# === Parameters:
+#
+# $pulpcore_url::       URL on which Pulpcore runs
+#
+# $username::           Username for authentication
+#
+# $password::           Password for authentication
+#
+# $cert::               Client certificate for authentication
+#
+# $key::                Client key for authentication
+#
+# === Advanced parameters:
+#
+# $manage_root_config:: Whether to manage /root/.config/pulp configuration.
+#
+# $api_root::           Absolute API base path on server (not including 'api/v3/')
+#
+# $verify_ssl::         Verify SSL connection
+#
+# $dry_run::            Trace commands without performing any unsafe HTTP calls
+#
+# $version::            pulp-cli package version, it's passed to ensure parameter of package resource
+#                       can be set to specific version number, 'latest', 'present' etc.
+#
+class pulpcore::cli (
+  Optional[Stdlib::HTTPUrl] $pulpcore_url = undef,
```

Comment:
should this default to `https://${fqdn}`?

Reply:
It's not exposed to users so I think this is fine.